GoogleUtilities/Example/Tests/Swizzler/GULAppDelegateSwizzlerTest.m

```diff
@@ -312,16 +346,181 @@ - (void)testProxyAppDelegate {
   // accessing the ivars should not crash.
   XCTAssertEqual(realAppDelegate->_arbitraryNumber, 123456789);
   XCTAssertEqual(realAppDelegate->_isInitialized, 1);
-  XCTAssertEqual(realAppDelegate->_isOpenURLOptionsMethodCalled, 0);
+  XCTAssertFalse(realAppDelegate->_isOpenURLOptionsMethodCalled);
   XCTAssertEqualObjects(realAppDelegate->_backgroundSessionID, @"randomSessionID");
 }
 
+- (void)testProxyEmptyAppDelegate {
+  GULEmptyTestAppDelegate *realAppDelegate = [[GULEmptyTestAppDelegate alloc] init];
+  OCMStub([self.mockSharedApplication delegate]).andReturn(realAppDelegate);
+  size_t sizeBefore = class_getInstanceSize([GULEmptyTestAppDelegate class]);
+
+  Class realAppDelegateClassBefore = [realAppDelegate class];
+
+  // Create the proxy.
+  [GULAppDelegateSwizzler proxyOriginalDelegate];
+
+  XCTAssertTrue([realAppDelegate isKindOfClass:[GULEmptyTestAppDelegate class]]);
+
+  NSString *newClassName = NSStringFromClass([realAppDelegate class]);
+  XCTAssertTrue([newClassName hasPrefix:@"GUL_"]);
+  // It is no longer GULTestAppDelegate class instance.
+  XCTAssertFalse([realAppDelegate isMemberOfClass:[GULEmptyTestAppDelegate class]]);
+
+  size_t sizeAfter = class_getInstanceSize([realAppDelegate class]);
+
+  // Class size must stay the same.
+  XCTAssertEqual(sizeBefore, sizeAfter);
+
+  // After being proxied, it should be able to respond to the required method selector.
+#if TARGET_OS_IOS
+  XCTAssertTrue([realAppDelegate
+      respondsToSelector:@selector(application:openURL:sourceApplication:annotation:)]);
+#endif  // TARGET_OS_IOS
+
+  XCTAssertTrue([realAppDelegate respondsToSelector:@selector(application:
+                                                        continueUserActivity:restorationHandler:)]);
+
+  // The implementation should not be added if there is no original implementation
+  XCTAssertFalse([realAppDelegate respondsToSelector:@selector(application:openURL:options:)]);
+  XCTAssertTrue([realAppDelegate
+      respondsToSelector:@selector(application:
+                             handleEventsForBackgroundURLSession:completionHandler:)]);
+
+  // Remote notifications methods should be added only by
+  // -proxyOriginalDelegateIncludingAPNSMethods
```

Comment:
> I might not be seeing it due to weird github diffs, but if it doesn't already exist, can you add a test that makes sure proxyOriginalDelegateIncludingAPNSMethods doesn't get swizzled when swizzling is disabled?

@tejasd It is tested here and in other places with such a comment

Reply:
This doesn't seem to test for situation where Swizzling is explicitly disabled, something like https://github.com/firebase/firebase-ios-sdk/blob/a4a94ce4643745f57b65c08624d0662eb0baf0f2/GoogleUtilities/Example/Tests/Swizzler/GULAppDelegateSwizzlerTest.m#L1157

Reply:
@tejasd  Ah, sorry, I misread the comment. I can add it. Do you think it is critical for this PR? 
`proxyOriginalDelegateIncludingAPNSMethods` uses `proxyOriginalDelegate` under the hood, worked as expected is implied with the current implementation, so it could be a separate PR

Reply:
Followup PR should be ok! Thanks!